crates/spl_network/src/endpoint.rs

```diff
@@ -90,7 +90,7 @@ impl Endpoint {
                     .await;
             }
             OutgoingMessage::Spl(message) => {
-                let message: Vec<u8> = message.into();
+                let message: Vec<u8> = bincode::serialize(&message).unwrap();
```

Comment:
When does this unwrap fail? Is it better to use an expect here with an error message?

Reply:
@Narcha

Reply:
also replaced with serde json

Reply:
We decided now to use `bincode` instead of `serde_json` because we are only allowed to use 128 bytes. Please convert the `unwrap` to an `except` with an appropriate error message. Also it would be a nice to write a test that asserts the size of the struct to be less or equal to 128 Bytes to prevent future additions of fields to break the rules.

Reply:
@Narcha  and I have changed this back to bincode and used a match statement to properly expect the error per @h3ndrk suggestions. Please feel free to comment if it should be different